comp/remote-config/rcclient/rcclientimpl/rcclient.go

```diff
@@ -291,14 +295,26 @@ func (rc rcClient) agentConfigUpdateCallback(updates map[string]state.RawConfig,
 
 		// Need to update the log level even if the level stays the same because we need to update the source
 		// Might be possible to add a check in deeper functions to avoid unnecessary work
-		err = rc.settingsComponent.SetRuntimeSetting("log_level", mergedConfig.LogLevel, model.SourceRC)
+		if err := rc.settingsComponent.SetRuntimeSetting("log_level", mergedConfig.LogLevel, model.SourceRC); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	streamLogsSource := config.Datadog().GetSource("logs_config.streaming.enable_streamlogs")
+	if mergedConfig.EnableStreamLogs {
+		if err := rc.settingsComponent.SetRuntimeSetting("enable_streamlogs", mergedConfig.EnableStreamLogs, model.SourceRC); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	} else if !mergedConfig.EnableStreamLogs && streamLogsSource == model.SourceRC {
+		config.Datadog().UnsetForSource("logs_config.streaming.enable_streamlogs", model.SourceRC)
 	}
```

Comment:
When is this condition triggered? By reading the code is not entirely clear when stream logs would not be enabled but the source is RC. 

Could you provide some context please?

Reply:
I'm not very familiar with how the settings from rc would be configured, but maybe @coignetp could clarify on this end 😅 My understanding is that we have a different config file on RC side that set/unset the value for different tasks.

Reply:
From RC side, we can send a `enable_stream_logs: true/false` so we can set the runtime settings (that would be done just like the log_level setting change)
It's totally fine to also set it from different parts of the code